cmake/version.cmake

```diff
@@ -2,28 +2,65 @@ FIND_PROGRAM(GIT_EXECUTABLE NAMES git git.exe DOC "git command line client")
 
 get_filename_component(SOURCE_DIR ${SRC} PATH)
 
-string(TIMESTAMP VERSION "%y%m%d%H%M")
+if (GIT_EXECUTABLE)
+   EXECUTE_PROCESS(
+      COMMAND ${GIT_EXECUTABLE} log "--format=%ct" # output as unix timestamp
+      WORKING_DIRECTORY "${SOURCE_DIR}"
+      OUTPUT_VARIABLE GIT_OUTPUT_DATE
+      ERROR_VARIABLE GIT_ERROR
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+      ERROR_STRIP_TRAILING_WHITESPACE
+   )
+   if(NOT GIT_OUTPUT_DATE)
+      message(STATUS "Cannot determine current git commit - git error: '${GIT_ERROR}'")
+      set(GIT_OUTPUT_DATE "0000")
```

Comment:
This doesn't seem to be consistent with the code on line 33, is that on purpose?

Reply:
oh yes, it seems that i missed that one. Even through it would not brake anything because the version code would be zero not relevant how much zeros you have